controllers/subscription_controller.go

```diff
@@ -433,3 +447,25 @@ func getNotUpgradeableCond(ocd *opv2.OperatorCondition) *metav1.Condition {
 		return cd.Type == opv2.Upgradeable && cd.Status != "True"
 	})
 }
+
+// Check if ODF version is n+1 compared to OCP version(e.g. OCP 4.20.0, ODF 4.21.0)
+func isODFAheadOfOCP(ctx context.Context, client client.Client, operatorConditionName string) (bool, error) {
+	ocpVersionString, err := util.DetermineOpenShiftVersion(ctx, client)
+	if err != nil {
+		return false, fmt.Errorf("failed to determine OCP version: %v", err)
+	}
+
+	ocpVersion, err := semver.Make(ocpVersionString)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse OCP version %s: %v", ocpVersionString, err)
+	}
+
+	// Parse ODF version from operatorConditionName as it's the same as the CSV name
+	version := operatorConditionName[strings.Index(operatorConditionName, ".v")+2:]
```

Comment:
better not, let's either take this as the actual csv name and get the version from it or traverse through the Pod -> Deployment -> CSV (iirc, being done in odf-op already at some place) to get the version.

I know this is easier but there are some variations in the csv name, sometimes it has `stable`, `rhodf` and just empty depending on environment and mostly doing name mangling on not established/not controlled entity (by us) is problematic.

Reply:
As per offline discussion now updated to get the csv with the name & then get the version from it.